core/src/components/range/range.tsx

```diff
@@ -73,11 +74,16 @@ export class Range implements ComponentInterface {
    * `ionChange` event after each change in the range value.
    * This also impacts form bindings such as `ngModel` or `v-model`.
    */
-  @Prop() debounce = 0;
+  @Prop() debounce?: number;
 
   @Watch('debounce')
   protected debounceChanged() {
-    this.ionChange = debounceEvent(this.ionChange, this.debounce);
+    const { ionChange, debounce, originalIonChange } = this;
+    /**
+     * If debounce is undefined, we have to manually revert the ionChange emitter in case
+     * debounce used to be set to a number. Otherwise, the event would stay debounced.
+     */
+    this.ionChange = debounce === undefined ? originalIonChange ?? ionChange : debounceEvent(ionChange, debounce);
```

Comment:
The Range component does not have an `ionInput` event to migrate the debounce implementation to. Should we still offer `debounce` here or alternatively, should we have an `ionInput` event that emits for each value change as the knob is dragged? 

The current `ionChange` emission pattern aligns with `<input type="range" />`, where the `change` event is not emitted until the knob is released or from a keypress.

Reply:
I think we should have an `ionInput` event otherwise developers may see this change as a regression since there currently is no way to know when the value changes while dragging.